os/drivers/testcase/testcase_drv.c

```diff
@@ -72,8 +73,142 @@ static int testdrv_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
 	 *   ioctl argument:  An integer value indicating the particular test to be run
 	 */
 
-	case TESTIOC_DRIVER_ANALOG: {
+	case TESTIOC_ANALOG: {
+
+	}
+	break;
+
+	case TESTIOC_CLOCK_ABSTIME2TICKS: {
+		base_time.tv_sec = (time_t)arg;
+		if (clock_abstime2ticks(CLOCK_REALTIME, &base_time, &ret) != OK) {
+			ret = ERROR;
+		}
+	}
+	break;
+
+	case TESTIOC_GET_SELF_TCB_PID: {
+		tcb =  sched_self();
+		if (tcb == NULL) {
+			ret = ERROR;
+			break;
+		}
+		ret = tcb->pid;
+	}
+	break;
+
+	case TESTIOC_GET_SIG_FINDACTION_ADD: {
+		ret = (int)sig_findaction(sched_self(), 2);
+		if (ret == 0) {
+			ret = ERROR;
+			break;
+		}
+	}
+	break;
+
+	case TESTIOC_GET_TCB_ALIVE: {
+		tcb = sched_gettcb((pid_t)arg);
+		if (tcb == NULL) {
+			ret = ERROR;
+			break;
+		}
+		ret = OK;
+	}
+	break;
+
+	case TESTIOC_GET_TCB_LOCKCOUNT: {
+		tcb = sched_gettcb((pid_t)arg);
+		if (tcb == NULL) {
+			ret = ERROR;
+			break;
+		}
+		ret = tcb->lockcount;
+	}
+	break;
+
+	case TESTIOC_GET_TCB_PID: {
```

Comment:
What is this? To use sched_gettcb, we must provide pid. But it finally returns pid !!!

Reply:
okay. it is strange. so i deleted it.